test/Raven/Tests/ReprSerializerTest.php

```diff
@@ -109,4 +109,45 @@ public function testSerializeRoundedFloat()
         $this->assertInternalType('string', $result);
         $this->assertEquals('12345.0', $result);
     }
+
+    /**
+     * @dataProvider serializableDataProvider
+     */
+    public function testSerializeSerializableObject($expected)
+    {
+        $serializer = new Raven_ReprSerializer();
+
+        $mock = $this->getMockBuilder('Raven_Serializable')
+                     ->getMock();
+        $mock->method('toSentry')->willReturn($expected);
+
+        $result = $serializer->serialize($mock);
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * @dataProvider serializableDataProvider
+     */
+    public function testGenericSerializableObject($expected)
+    {
+        $serializer = new Raven_ReprSerializer();
+
+        $mock = $this->getMockBuilder('stdClass')
+                     ->setMethods(array('toDebugContext'))
+                     ->getMock();
+        $mock->method('toDebugContext')->willReturn($expected);
+
+        $result = $serializer->serialize($mock);
+        $this->assertEquals($expected, $result);
```

Comment:
I guess the problem here is the fact that serializeValue is used for serializing array's keys. However the value can be a simple array and I think would be useful to allow it for more meaningfull representation.

Wouldn't be better to allow array in serializeValue and use serializeString for array's key?